process/transaction/transactionCostEstimator.go

```diff
@@ -134,6 +172,37 @@ func (tce *transactionCostEstimator) addMissingFieldsIfNeeded(tx *transaction.Tr
 	}
 }
 
+func (tce *transactionCostEstimator) getTxGasLimit(txSender []byte) (gasLimit uint64) {
+	selfShardID := tce.shardCoordinator.SelfId()
+	maxGasLimitPerBlock := tce.feeHandler.MaxGasLimitPerBlock(selfShardID) - 1
+
+	gasLimit = maxGasLimitPerBlock
+
+	senderShardID := tce.shardCoordinator.ComputeId(txSender)
+	if tce.shardCoordinator.SelfId() != senderShardID {
+		return
+	}
+
+	accountHandler, err := tce.accounts.LoadAccount(txSender)
+	if err != nil || accountHandler == nil {
+		return
+	}
+
+	accountSender, ok := accountHandler.(state.UserAccountHandler)
+	if !ok {
+		return
+	}
+
+	maxGasLimitPerBlockBig := big.NewInt(0).SetUint64(maxGasLimitPerBlock)
+	accountSenderBalance := accountSender.GetBalance()
+	if accountSenderBalance.Cmp(maxGasLimitPerBlockBig) < 0 && accountSenderBalance.Uint64() != 0 {
```

Comment:
this is not correct. You compare gas limit with balance.

Reply:
here you have to call economics.ComputeTxFee

Reply:
done.